contracts/Nitro-SCW.sol

```diff
@@ -49,9 +49,11 @@ contract NitroSmartContractWallet is IAccount {
         removeActiveHTLC(hashLock);
         updateLatestExpiry();
 
-        // Only tranfer funds if they're targetted outside of this wallet
-        if (htlc.to != address(this)) {
-            htlc.to.transfer(htlc.amount);
+        if (htlc.to == intermediary) {
+            intermediaryBalance += htlc.amount;
+        }
+        if (htlc.to == owner) {
+            intermediaryBalance -= htlc.amount;
         }
 
```

Comment:
```suggestion

```

If there's an HTLC with balance 2 that points to the owner, then the addition of that HTLC to the state should have already done the decrement for intermediaryBalance.

Reply:
Hmm, so what can Alice(the owner) do with the secret preimage revealed by Irene(the intermediary) to unlock the HTLC (paying to Alice)? If this `UnlockHTLC` function only handles HTLCs targetting the intermediary doesn't that mean there's no on-chain way to make use of that preimage?

Reply:
The `preimage` is still used to remove the `HTLC` (even if it doesn't change `intermediaryBalance`) so it's still important, because that means more funds will remain in the wallet.